25_exceptions_lab/03.py

```diff
@@ -1,4 +1,36 @@
 """
 Write code to make the following unit test pass
 """
+import re
+import unittest
 
+class InvalidImageExt(Exception):
+    def __init__(self, message):
+        self._message = message
+    
+    def __str__(self):
+        return self._message
+
+class ImageFile(object):
+    validExtension = "png"
+    
+    def __init__(self, fileName):
+        result = re.search("[a-z,0-9]*.(?P<ext>[a-z,0-9]+)", fileName)
```

Comment:
Python has an `os.path.basename` function that does a better job than writing your own regexp